nips/05.md

```diff
@@ -0,0 +1,52 @@
+NIP-05
+======
+
+Mapping Nostr keys to DNS-based internet identifiers
+----------------------------------------------------
+
+`draft` `optional` `author:fiatjaf`
+
+On events of type `0` (`set_metadata`) one can specify the key `"nip05"` with an [internet identifier](https://datatracker.ietf.org/doc/html/rfc5322#section-3.4.1) (an email-like address) as the value. Although there is a link to a very liberal "internet identifier" specification above, NIP-05 assumes the `<local-part>` part will be restricted to the characters `a-z0-9-_.`, case insensitive.
```

Comment:
Should client also do validation when creating new `set_metadata` event?

And what about receiving relay? Should it validate reveived events with `nip05` key?

Reply:
You mean clients should validate if the specified nip05 field really translates into that same pubkey when the server is queried?

Yes, I guess they could do that, but I personally don't think it's a good idea.